Functions/xASL_quant_Basil.m

```diff
@@ -123,170 +130,203 @@
-     fprintf(FIDoptionFile, '--slicedt=%f\n', x.Q.BasilSliceReadoutTime/1000);
-     fprintf(FIDoptionFile, '--save-model-fit\n');
-
-
-    %% FIXME Aquisition options we might be able to use in the future
-    %fprintf(option_file, '--FA=%f\n', fa);
-    %fprintf('Basil: Flip angle for look-locker readout: %f\n', fa);
-    %fprintf(option_file, '--sliceband=%i\n', sliceband);
-    %fprintf('Basil: Multi-band setup with number of slices per band: %i\n', slicedband);
-
-    % This helps avoid failure on the structural-space image
-    fprintf(FIDoptionFile, '--allow-bad-voxels\n');
-
-    %% FIXME is a user-specified T1 map possible in ExploreASL?
-    %fprintf(option_file, '--t1im=%s\n', t1im)
-    %fprintf('Basil: Using supplied T1 (tissue) image in BASIL: %s\n', $t1im)
-
-    % Labelling type - PASL or pCASL
-    switch lower(x.Q.LabelingType)
-        case 'pasl'
-            fprintf('Basil: PASL model\n');
-        case 'casl'
-            fprintf(FIDoptionFile, '--casl\n');
-            fprintf('Basil: CASL/PcASL model\n');
-    end
-
-
-    %% Model option - 1 or 2 compartment
-    % The 1-compartment model is what we call 'White paper mode' in oxford-asl.
-    % This means zero ATT (all bolus delivered by imaging time) and blood T1 only
-    % The 2-compartment model is the 'standard' buxton model which takes into account
-    % the ATT and tissue T1 value.
-
-    switch x.Q.nCompartments
-        case 1
-            fprintf(FIDoptionFile, '--bat=0\n');
-            fprintf(FIDoptionFile, '--t1=%f\n', x.Q.BloodT1/1000);
-            fprintf('Basil: Single-compartment (white paper mode)\n');
-        case 2
-            fprintf(FIDoptionFile, '--bat=%f\n', x.Q.ATT/1000);
-            fprintf(FIDoptionFile, '--t1=%f\n', x.Q.TissueT1/1000);
-            fprintf('Basil: 2-compartment - ATT=%fs\n', x.Q.ATT/1000);
-    end
-
-    
-    %%  Bolus duration\labeling duration typically fixed but can be inferred
-    if isfield(x.Q,'BasilInferTau') && x.Q.BasilInferTau
-         fprintf(FIDoptionFile, '--infertau\n');
-         fprintf('Basil: Infer bolus duration component\n')
-    else
-        fprintf('Basil: Fixed bolus duration component\n')
-    end
-
-    
-    %% ATT and arterial component inference only possible with multi-PLD
-    if ~bMultiPLD
-        fprintf('Basil: Single-delay data - cannot infer ATT or arterial component\n');
-        x.Q.BasilInferATT = 0;
-        x.Q.BasilInferArt = 0;
-    end
+% Right now, we assume that we have averaged over PLDs
+%fprintf(FIDoptionFile, '--repeats=%i\n', size(PWI, 4)/PLDAmount);
+fprintf(FIDoptionFile, '--repeats=1\n');
+	
+% Slice-timing
+fprintf(FIDoptionFile, '--slicedt=%f\n', x.Q.BasilSliceReadoutTime/1000);
+
+if isfield(x.Q,'LookLocker') && x.Q.LookLocker 
+	if isfield(x.Q,'FlipAngle')
+		fprintf(option_file, '--FA=%f\n', x.Q.FlipAngle);
+		fprintf('BASIL: Flip angle for Look-Locker readout: %f\n', x.Q.FlipAngle);
```

Comment:
I know of only fixed FA Look-Locker, but is it possible to have multiple FAs? Would complicate the reconstruction I would say, just curious

Reply:
Good point. Look-Locker practically have a single-FA only per volume (or a single volume batch after a single labeling), but you can indeed have different FAs for different repetitions - like in QUASAR. Currently, BASIL seems to accept only a single FA value here, and QUASAR has it's own quantification function. So since almost nobody uses Look-Locker, and nobody uses QUASAR, no point implementing QUASAR now and we can leave it here for single FA. I added a warning though: 4dffd3ca